src/usernotificationsui.cs

```diff
@@ -0,0 +1,93 @@
+﻿//
+// UserNotificationsUI bindings
+//
+// Authors:
+//	Alex Soto  <alex.soto@xamarin.com>
+//
+// Copyright 2016 Xamarin Inc. All rights reserved.
+//
+
+using System;
+using XamCore.CoreGraphics;
+using XamCore.Foundation;
+using XamCore.ObjCRuntime;
+using XamCore.UIKit;
+using XamCore.UserNotifications;
+
+namespace XamCore.UserNotificationsUI {
+
+	[Introduced (PlatformName.iOS, 10, 0)]
+	[Unavailable (PlatformName.MacOSX)]
+	[Unavailable (PlatformName.WatchOS)]
+	[Unavailable (PlatformName.TvOS)]
+	[Native]
+	public enum UNMediaPlayPauseButtonType : nuint {
+		None,
+		Audio,
+		Movie
+	}
+
+	[Introduced (PlatformName.iOS, 10, 0)]
+	[Unavailable (PlatformName.MacOSX)]
+	[Unavailable (PlatformName.WatchOS)]
+	[Unavailable (PlatformName.TvOS)]
+	[Native]
+	public enum UNResponseOption : nuint {
+		DoNotDismiss,
+		Dismiss,
+		DismissAndForwardAction
+	}
+
+	interface IUNNotificationContentExtension { }
+
+	[Introduced (PlatformName.iOS, 10, 0)]
+	[Unavailable (PlatformName.MacOSX)]
+	[Unavailable (PlatformName.WatchOS)]
+	[Unavailable (PlatformName.TvOS)]
+	[Protocol]
+	[BaseType (typeof (NSObject))]
+	interface UNNotificationContentExtension {
+
+		[Abstract]
+		[Export ("didReceiveNotification:")]
+		void DidReceiveNotification (UNNotification notification);
+
+		[Export ("didReceiveNotificationResponse:completionHandler:")]
+		void DidReceiveNotificationResponse (UNNotificationResponse response, Action<UNResponseOption> completion);
+
+		[Export ("mediaPlayPauseButtonType", ArgumentSemantic.Assign)]
+		UNMediaPlayPauseButtonType MediaPlayPauseButtonType { get; }
+
+		[Export ("mediaPlayPauseButtonFrame", ArgumentSemantic.Assign)]
+		CGRect MediaPlayPauseButtonFrame { get; }
+
+		[Export ("mediaPlayPauseButtonTintColor", ArgumentSemantic.Copy)]
+		UIColor MediaPlayPauseButtonTintColor { get; }
+
+		[Export ("mediaPlay")]
+		void MediaPlay ();
+
+		[Export ("mediaPause")]
+		void MediaPause ();
+	}
+
+	[Category]
+	[BaseType (typeof (NSExtensionContext))]
+	interface NSExtensionContext_UNNotificationContentExtension {
+
+		[Introduced (PlatformName.iOS, 10, 0)]
+		[Unavailable (PlatformName.MacOSX)]
+		[Unavailable (PlatformName.WatchOS)]
+		[Unavailable (PlatformName.TvOS)]
```

Comment:
Can't those be moved to the type itself, instead of having them defined twice inside the category ?